includes/class-wcsatt-order.php

```diff
@@ -0,0 +1,153 @@
+<?php
+/**
+ * WCS_ATT_Order class
+ *
+ * @author   SomewhereWarm <info@somewherewarm.gr>
+ * @package  WooCommerce Subscribe All the Things
+ * @since    2.0.0
+ */
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Order hooks for saving/restoring the subscription state of a product to/from order item data.
+ *
+ * @class    WCS_ATT_Order
+ * @version  2.0.0
+ */
+class WCS_ATT_Order {
+
+	public static function init() {
+
+		// Restore subscription data when creating a cart item using an order item as reference.
+		add_filter( 'woocommerce_order_again_cart_item_data', array( __CLASS__, 'restore_cart_item_from_order_item' ), 10, 3 );
+
+		// Restore the subscription state of a product instantiated using an order item as reference.
+		if ( WCS_ATT_Core_Compatibility::is_wc_version_gte_2_7() ) {
+			add_filter( 'woocommerce_order_item_product', array( __CLASS__, 'restore_product_from_order_item' ), 10, 2 );
+		} else {
+			// Using this under WC 3.0 may result in serious performance issues.
+			add_filter( 'woocommerce_get_product_from_item', array( __CLASS__, 'restore_product_from_order_item' ), 10, 2 );
+		}
+
+		// Save subscription scheme in subscription item meta so it can be re-applied later.
+		if ( WCS_ATT_Core_Compatibility::is_wc_version_gte_2_7() ) {
+			add_action( 'woocommerce_checkout_create_order_line_item', array( __CLASS__, 'save_subscription_scheme_meta' ), 10, 3 );
+		} else {
+			add_action( 'woocommerce_add_order_item_meta', array( __CLASS__, 'save_subscription_scheme_meta_legacy' ), 10, 2 );
+		}
+	}
+
+	/*
+	|--------------------------------------------------------------------------
+	| API
+	|--------------------------------------------------------------------------
+	*/
+
+	/**
+	 * Returns the key of the subscription scheme applied on the product when it was purchased.
+	 *
+	 * @param  array  $order_item
+	 * @return string|false|null
+	 */
+	public static function get_subscription_scheme( $order_item ) {
```

Comment:
IMO, getters which explicitly mention the expected return value reduce ambiguity so are preferred. But I'll leave the decision whether to change both `WCS_ATT_Product::get_subscription_scheme` and `WCS_ATT_Cart::get_subscription_scheme` up to you. :)